burn-core/src/nn/loss/mse.rs

```diff
@@ -0,0 +1,81 @@
+use core::marker::PhantomData;
+
+use burn_tensor::{backend::Backend, Tensor};
+
+/// Calculate the mean squared error loss from the input logits and the targets.
+#[derive(Clone, Debug)]
+pub struct MSELoss<B: Backend> {
+    backend: PhantomData<B>,
+}
+
+impl<B: Backend> Default for MSELoss<B> {
+    fn default() -> Self {
+        Self::new()
+    }
+}
+
+impl<B: Backend> MSELoss<B> {
+    /// Create the criterion.
+    pub fn new() -> Self {
+        Self {
+            backend: PhantomData::default(),
+        }
+    }
+
+    /// Compute the criterion on the input tensor.
+    ///
+    /// # Shapes
+    ///
+    /// - logits: [batch_size, num_targets]
+    /// - targets: [batch_size, num_targets]
```

Comment:
I think it would be better to have the reduction handled by an enum here, so there would be only one forward function.

```rust
pub fn forward(&self, logits: Tensor<B, 2>, targets: Tensor<B, 2>, reduction: Reduction) -> Tensor<B, 1> {
    match reduction {
        Reduction::Meam => ...
        Reduction::Sum => ...
    }
}

pub enum Reduction {
   Mean,
   Sum,
}
```

Since the return type is different when no reduction is applied, I would keep the `forward_none`, but rename it to `forward_no_reduction`.